src/grid/initial_conditions.cpp

```diff
@@ -291,6 +297,61 @@ void Grid3D::Sound_Wave(Real rho, Real vx, Real vy, Real vz, Real P, Real A)
         #ifdef DE
 	C.GasEnergy[id]  = P/(gama-1.0);
         #endif  //DE
+        #ifdef DE
+	C.GasEnergy[id]  = P/(gama-1.0);
+        #endif  //DE
+      }
+    }
+  }
+
+}
+
+/*! \fn void Linear_Wave(Real rho, Real vx, Real vy, Real vz, Real P, Real A)
+ *  \brief Sine wave perturbation. */
+void Grid3D::Linear_Wave(Real rho, Real vx, Real vy, Real vz, Real P, Real A,
+                        Real Bx, Real By, Real Bz, Real rEigenVec_rho,
+                        Real rEigenVec_MomentumX, Real rEigenVec_MomentumY, Real rEigenVec_MomentumZ,
+                        Real rEigenVec_E, Real rEigenVec_Bx, Real rEigenVec_By,
+                        Real rEigenVec_Bz, Real pitch, Real yaw)
+{
+  auto [stagger, junk1, junk2] = math_utils::rotateCoords<Real>(H.dx/2, H.dy/2, H.dz/2, pitch, yaw);
+
+  // set initial values of conserved variables
+  for(int k=H.n_ghost; k<H.nz-H.n_ghost; k++) {
+    for(int j=H.n_ghost; j<H.ny-H.n_ghost; j++) {
+      for(int i=H.n_ghost; i<H.nx-H.n_ghost; i++) {
+        // Rotate the indices
+        auto [i_rot, j_rot, k_rot] = math_utils::rotateCoords<int>(i, j, k, pitch, yaw);
+
+        //get cell index
+        int id = i + j*H.nx + k*H.nx*H.ny;
+
+        // get cell-centered position
+        Real x_pos, y_pos, z_pos;
+        Get_Position(i_rot, j_rot, k_rot, &x_pos, &y_pos, &z_pos);
+
+        // set constant initial states. Note that mhd::utils::computeEnergy
+        // computes the hydro energy if MHD is turned off
+        Real sine_wave = std::sin(2.0 * PI * x_pos);
+
+        C.density[id]    = rho;
+        C.momentum_x[id] = rho*vx;
+        C.momentum_y[id] = rho*vy;
+        C.momentum_z[id] = rho*vz;
+        C.Energy[id]     = mhd::utils::computeEnergy(P, rho, vx, vy, vz, Bx, By, Bz, gama);
```

Comment:
Only this function is compiled if MHD is turned off. `mhd::utils::computeEnergy` returns the correct hydro or mhd energy depending on what flags are enabled. IMO we really need to unify these utility functions. We should be able to just call "computeEnergy" or something similar and it will automatically choose the correct way to compute the energy, then there can be separate functions if you need a specific one. Otherwise at some point someone is going to be computing the hydro energy when they want the MHD energy or vice versa